src/adapter/src/catalog.rs

```diff
@@ -6498,6 +6626,14 @@ impl mz_sql::catalog::CatalogRole for Role {
     fn create_cluster(&self) -> bool {
         self.attributes.create_cluster
     }
+
+    fn membership(&self) -> BTreeSet<&RoleId> {
```

Comment:
nit: maybe just return an iterator here to avoid the allocation for callers that don't need a set?

Reply:
When I do this `CatalogRole` is no longer object safe which means that we can't use it in `&dyn CatalogRole` contexts (I think). That probably makes this not worth it, unless you have some advice here.

Reply:
Ahhh makes sense, I missed that this was part of a trait. Leaving it as-is sounds good then